nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -581,30 +585,31 @@ private Status getStatus(final Logger logger) {
         return new Status(port, pid, pingSuccess, alive);
     }
 
-    public void status() throws IOException {
+    public int status() throws IOException {
         final Logger logger = cmdLogger;
         final Status status = getStatus(logger);
         if (status.isRespondingToPing()) {
             logger.info("Apache NiFi is currently running, listening to Bootstrap on port {}, PID={}",
                     new Object[]{status.getPort(), status.getPid() == null ? "unknown" : status.getPid()});
-            return;
+            return 0;
         }
 
         if (status.isProcessRunning()) {
             logger.info("Apache NiFi is running at PID {} but is not responding to ping requests", status.getPid());
-            return;
+            return 0;
```

Comment:
there is a PID, it is listed in list of running processes... it may be busy and not respong to probes, but the process itself is in state running, at least kernel thinks so. it may be in deadlock, may it be starvating, but it is running.